### DISCUSSION TITLE
Blocking sub-domains - broken syntax?

### DISCUSSION THREAD

**Discussion by EntropySmoke** [AdguardTeam/AdGuardHome#3827] [Q&A]
Running the latest Edge of AGH on RP2 Bullseye and using AGH to block all domains, except the ones needed to connect clients to VPN tunnels.
Example issue:
||*^$client=10.0.0.1
@@||fakedomain.com^$important
||subdomain.fakedomain.com^$important

With above rules I expect AGH to block all domains for client 10.0.0.1, unless the domain is fakedomain.com, but also block subdomain.fakedomain.com. AGH doesn't block subdomain.fakedomain.com. What is the correct syntax?

**Comment by ainar-g** [comment]
1.  You should use the `client` modifier for all rules that work for a client.

1.  Exception rules have a higher priority.

Considering these, you should be able to accomplish what you want with the following set of rules:

```none
||*^$client=10.0.0.1
@@||google.com^$client=10.0.0.1
||mail.google.com^$client=10.0.0.1,important
```

**Comment by EntropySmoke** [comment]
Got it! Thank you!
